### ISSUE TITLE
RCS1095 False Negative

### ISSUE THREAD

**Issue by nnpcYvIVl** [JosefPihrt/Roslynator#88]
Dictionary1 throws RCS1095 "Use C# 6.0 initializer" but Dictionary2 doesn't. Can you make RCS1095 work for classes that inherit from Dictionary?

```csharp
public class NewDictionary<T, U> : Dictionary<T, U> { }

public class Class1
{
    Dictionary<string, string> Dictionary1 = new Dictionary<string, string>()
    {
        {"a", "b"},
    };

    NewDictionary<string, string> Dictionary2 = new NewDictionary<string, string>()
    {
        {"a", "b"},
    };
}
```
